frontend/test/jest-setup.js

```diff
@@ -11,6 +12,12 @@ process.on("uncaughtException", err =>
   console.error("WARNING: UNCAUGHT EXCEPTION", err),
 );
 
+if (!global.fetch) {
+  global.fetch = fetch;
+  global.Request = Request;
+  global.Response = Response;
+}
+
```

Comment:
Am I right that we're now using node jest environment, and here we patch up the `global` node object to add `fetch` there? If so:

1. Don't we need to change [the `testEnvironment` in jest config](https://github.com/metabase/metabase/blob/464ec9e5bede2aec50391d50f8f7b5b600e61f96/jest.unit.conf.json#L52)?
2. We won't need `node-fetch` once we switch to a Node version with built-in `fetch` on CI, right?

Reply:
`node` doesn't support `fetch` at all, it's browser-only

Reply:
here I used `node-fetch` that implements `fetch` using the built-in `http` package for `node` because `nock` only works with `http`. If we switch from `nock` to something else we would use a different approach

Reply:
Got it. I meant Node 18 has [built-in `fetch` API](https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#fetch) (it's experimental though)